python/lsst/daf/butler/core/utils.py

```diff
@@ -475,3 +478,58 @@ def isplit(string: T, sep: T) -> Iterator[T]:
             return
         yield string[begin:end]
         begin = end + 1
+
+
+@contextmanager
+def time_this(log: Optional[logging.Logger] = None, msg: Optional[str] = None,
+              log_level: int = logging.DEBUG, log_prefix: Optional[str] = None,
+              args: Iterable[Any] = ()) -> Iterator[None]:
+    """Time the enclosed block and issue a log message.
+
+    Parameters
+    ----------
+    log : `logging.Logger`, optional
+        Logger to use to report the timer message. The root logger will
+        be used if none is given.
+    msg : `str`, optional
+        Context to include in log message.
+    log_level : `int`, optional
+        Python logging level to use to issue the log message. If the
+        code block raises an exception the log message will automatically
+        switch to level ERROR.
+    log_prefix : `str`, optional
```

Comment:
Your invocations are verbose enough that I wonder if you want to simplify by defaulting to prefixing with `timer`.

Reply:
Okay. I'll default it to `timer`. I was also wondering about dropping the `log_` from the variable names so maybe I should do that as well.